src/client/lazy-app/Compress/Modal/index.tsx

```diff
@@ -0,0 +1,148 @@
+import { h, Component } from 'preact';
+import * as style from './style.css';
+import 'add-css:./style.css';
+import { linkRef } from 'shared/prerendered-app/util';
+import {
+  InfoIcon,
+  DiamondStarIcon,
+  ModalErrorIcon,
+} from 'client/lazy-app/icons';
+
+interface Props {}
+
+export interface ModalMessage {
+  type: 'info' | 'error' | 'update';
+  title: string;
+  content: string;
+}
+
+interface State {
+  message: ModalMessage;
+  shown: boolean;
+}
+
+export default class Modal extends Component<Props, State> {
+  state: State = {
+    message: {
+      type: 'info',
+      title: 'default',
+      content: 'default',
+    },
+    shown: false,
+  };
+
+  private modal?: HTMLElement;
+  private returnFocusElement?: HTMLElement | null;
```

Comment:
Not required, but we can start using 'real' private properties. If you want to do that, you might need to adjust the config.

The benefit in real private vars is their names compress.

Reply:
Ahh got it, will do